server/src/main/java/org/apache/cloudstack/consoleproxy/ConsoleAccessManagerImpl.java

```diff
@@ -207,15 +199,22 @@ private ConsoleEndpoint generateAccessEndpoint(Long vmId, String sessionUuid, St
             throw new CloudRuntimeException(msg);
         }
 
+        String vmUuid = vm.getUuid();
+        if (vm.getState() != VirtualMachine.State.Running) {
```

Comment:
@nvazquez 
is it possible to allow the console for vms in Running/Starting/Stopping/Migrating states ?

Reply:
For Running it is, do you think Starting, Stopping and Migrating should be required as well?

Reply:
@nvazquez 
I think so.
the console icon is previously displayed for Starting/Stopping/Migrating VMs on UI.(I do not know the current behavior)

for Stopping/Migrating VMs, console works mostly (please notice the Migrating vm is running on the last host, not current host)
for Starting VMs, sometimes console works, sometimes console does not work (depend on the VM state on hypervisor).

You can test with 4.17

Reply:
https://github.com/apache/cloudstack/blob/main/ui/src/components/widgets/Console.vue#L22 we show the console when VM state is not in error, destroyed or stopped ?

Reply:
Thanks @weizhouapache @rohityadavcloud, I'll update accordingly